### ISSUE TITLE
make: *** No rule to make target `requirements-test.txt', needed by `venv-test/touchfile'.  Stop.

### ISSUE THREAD

**Issue by weiweihuanghuang** [googlefonts/googlefonts-project-template#176]
I'm trying to update my [Fragment Mono](https://github.com/weiweihuanghuang/fragment-mono) to the latest googlefonts-project-template. So I run `make update-project-template` then I run `make update` and I get this error:

```
test -d venv || python3 -m venv venv
. venv/bin/activate; pip install -Ur requirements.txt
ERROR: Could not open requirements file: [Errno 2] No such file or directory: 'requirements.in'
```

So I grab the `requirements.in` from the [latest comit](https://github.com/googlefonts/googlefonts-project-template/tree/ca4b21bee0745406b28c1a647d02802634e03a1a) and run it again which updates a bunch of tools then I get this message:

```
touch venv/touchfile
make: *** No rule to make target `requirements-test.txt', needed by `venv-test/touchfile'.  Stop.
```

What's that mean?

**Comment by simoncozens** [comment]
I'm not surprised by the pip thing. The versions of pip when you run it on the command line and the version installed in the venv are different.

Obviously `make update-project-template` is not giving you the files which have been created since you used the template, which is a pain. Unfortunately I don't know a better way of forcing an update, but I will see if I can find something.

For pycairo to build, looks like you need `brew upgrade libffi`.

**Comment by weiweihuanghuang** [comment]
Can I manually make the update somehow?

**Comment by simoncozens** [comment]
You don't need to update pip, it should be good enough; you've done the manual update of the template by copying the new files across; you just now need to update libffi.

**Comment by weiweihuanghuang** [comment]
Ok I wiped reset my project folder and ran the above steps (including copying requirements.in, requirements-test.in, requirements-test.txt) and get a different error now:

```
touch venv/touchfile
test -d venv-test || python3 -m venv venv-test
. venv-test/bin/activate; pip install -Ur requirements-test.txt
/bin/sh: venv-test/bin/activate: No such file or directory
make: *** [venv-test/touchfile] Error 1
```

**Comment by weiweihuanghuang** [comment]
The only way I could get it to run `make update` was by cloning the repo fresh and then running `make customize` then manually copying the `venv` folder over. Then I get this error:

```
venv/bin/pip install --upgrade pip-tools
make: venv/bin/pip: No such file or directory
make: *** [update] Error 1
```

**Comment by weiweihuanghuang** [comment]
If there's no problem with me manually copying the updates over then can I just push that as my project comit?

**Comment by janbrasna** [comment]
The main dependency woes come from mismatching Python versions.

The template requirements at the time were expecting Python 3.10, not 3.11 as seen in the logs above. (Analogously today the stack's updated to 3.11, but consumers might run into rough edges trying to work e.g. on 3.12 or 3.14 environments…)

It's also good to nuke any venvs when upgrading to start fresh.

(I'm going to add some of the above to docs, to have a reminder there.)

1. Yes, it's actually more transparent to superimpose the template files manually, to see also e.g. removals to decide if that's relevant to one's project too.
1. `vent-test` is now gone so this no longer applies…
1. Good point `requirements.in` is actually missing from the `.templaterc` — I'll fold that into some other fixes when in the neighborhood.
1. I see you now upgraded successfully — I believe this can be closed now?

**Comment by weiweihuanghuang** [comment]
Yes I updated successfully but in the end I just manually did it, I wasn't sure what it was actually doing every time I updated.
